File: front-end/src/Services/BookService.js
import axios from 'axios';

const Book_API_BASE_URL = "http://localhost:8080/Books";

class BookServices {

    getBook(){
        return axios.get(Book_API_BASE_URL);
    }

    createBook(book){
        return axios.post(Book_API_BASE_URL, book);
    }

    getBookById(bookId){
        return axios.get(Book_API_BASE_URL + '/' + bookId);
    }

    updateBook(book, bookId){
        return axios.put(Book_API_BASE_URL + '/' + bookId, book);
    }

    deleteBook(bookId){
        return axios.delete(Book_API_BASE_URL + '/' + bookId);
    }
}

export default new BookServices()

File: Book/src/main/java/com/hexaware/bookstoremanagement/service/BookService.java
package com.hexaware.bookstoremanagement.service;

import com.hexaware.bookstoremanagement.entity.Book;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface BookService {
    public List<Book> getAllBooks();
    public int getBook(int cust_id);
    public Book createBook(Book book);
    public ResponseEntity<Book> updateBook(int book_id, Book book);
    public void deleteBook(int book_id);

    String getBookName(int id);
}